packages/pyright-internal/src/analyzer/importResolver.ts

```diff
@@ -531,12 +531,20 @@ export class ImportResolver {
             const entries = this.fileSystem.readdirEntriesSync(uri);
             entries.forEach((entry) => {
                 newCachedDir.entries.set(entry.name, entry);
-                const resolvableName = entry.isFile()
+                const isFile =
+                    entry.isFile() ||
+                    (entry.isSymbolicLink() && tryStat(this.fileSystem, uri.combinePaths(entry.name))?.isFile());
```

Comment:
The expression `entry.isSymbolicLink() && tryStat(this.fileSystem, uri.combinePaths(entry.name)` is evaluated twice under some circumstances. Not only does this involve repeated logic, but it also results in duplicate work at runtime. This is a hot path, so we should avoid making unnecessary system calls (`tryStat`). I recommend restructuring the code so this is called only once.